web/modules/custom/collection/src/Plugin/Block/JoinCollectionBlock.php

```diff
@@ -56,12 +57,13 @@ class JoinCollectionBlock extends BlockBase implements ContainerFactoryPluginInt
    * @param \Drupal\Core\Session\AccountProxyInterface $user
    *   The current user.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $current_route_match, AccountProxyInterface $user) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $current_route_match, AccountProxyInterface $user, ContextProviderInterface $collection_context) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
     $this->currentRouteMatch = $current_route_match;
     $this->user = $user;
-    // Retrieve the collection from the route.
-    $this->collection = $this->currentRouteMatch->getParameter('rdf_entity');
+    if (!empty($collection_context->getRuntimeContexts(['collection'])['collection']->getContextValue())) {
+      $this->collection = $collection_context->getRuntimeContexts(['collection'])['collection']->getContextValue();
+    }
```

Comment:
The context parameter is in the condition of the block. If I throw an exception here, the installation of the profile fails because when it loads the configuration for the first time and tries to save the block, the constructor is fired (I guess) and the exception is thrown! So the installation fails.
Two things to notice here, first, if the `->getContextValue();` from this line has an issue, it will fire an exception (I removed the check) and it does not fire during the installation of the profile.
Second, we anyway include a check with an exception during at the build method of this block class. We are already covered for showing in pages with the collection context.